Mathlib/Data/Multiset/Basic.lean

```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro
 
 ! This file was ported from Lean 3 source module data.multiset.basic
-! leanprover-community/mathlib commit f3187269ad18e82a809428a42d6282ce81e4ebcc
+! leanprover-community/mathlib commit 47adfab39a11a072db552f47594bf8ed2cf8a722
```

Comment:
The changes that remain here are left from leanprover-community/mathlib#18127 and leanprover-community/mathlib#18181, both of which are huge PRs.

I'm not entirely sure how to best review them.

Reply:
For now I would recommend splitting this PR into one PR for `finset` (which has only trivial changes and is easy to merge), and one for `multiset` which we likely want to think about better tooling before merging.

Reply:
The changes to `Data.Finset.Basic` have been merged as #2657 so all that remains here is syncing `Data.Multiset.Basic`